File: spaceship-cognito/app/src/main/java/it/scoppelletti/spaceship/cognito/sample/MainApp.java
package it.scoppelletti.spaceship.cognito.sample;

import android.app.Application;
import android.content.Context;
import com.amazonaws.regions.Regions;
import org.greenrobot.eventbus.EventBus;
import it.scoppelletti.spaceship.cognito.CognitoAdapter;

public final class MainApp extends Application {

    public static final String TAG_MAINACTIVITYDATA =
            "it.scoppelletti.spaceship.cognito.sample.1";

    @Override
    public void onCreate() {
        Context ctx;

        super.onCreate();

        EventBus.builder()
                .sendSubscriberExceptionEvent(false)
                .sendNoSubscriberEvent(false)
                .eventInheritance(false)
                .throwSubscriberException(BuildConfig.DEBUG)
                .addIndex(new MainEventBusIndex())
                .installDefaultEventBus();

        ctx = getApplicationContext();
        new CognitoAdapter.Builder(ctx)
                .poolId(BuildConfig.POOL_ID)
                .clientId(BuildConfig.CLIENT_ID)
                .clientSecret(BuildConfig.CLIENT_SECRET)
                .region(Regions.EU_WEST_1).build();
    }
}

File: spaceship-html/app/src/main/kotlin/it/scoppelletti/spaceship/html/sample/MainApp.kt
package it.scoppelletti.spaceship.html.sample

import android.app.Application
import it.scoppelletti.spaceship.html.sample.inject.AppComponent
import it.scoppelletti.spaceship.html.sample.inject.DaggerAppComponent
import it.scoppelletti.spaceship.inject.StdlibComponent
import it.scoppelletti.spaceship.inject.StdlibComponentProvider
import it.scoppelletti.spaceship.inject.UIComponent
import it.scoppelletti.spaceship.inject.UIComponentProvider

@Suppress("unused")
class MainApp : Application(), StdlibComponentProvider, UIComponentProvider {

    private lateinit var _appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()

        _appComponent = DaggerAppComponent.factory()
                .create(this)
    }

    override fun stdlibComponent(): StdlibComponent = _appComponent

    override fun uiComponent(): UIComponent = _appComponent
}
